### PR TITLE
Add spelling correction for sycle and variants.

### PR DESCRIPTION
**Pull request by cfi-gb**
See e.g.:

https://grep.app/search?q=sycle&words=true
https://grep.app/search?q=sycles&words=true (https://github.com/search?q=sycles&type=code has more hits)
https://github.com/search?q=sircular&type=code
https://github.com/search?q=syclical&type=code

There is https://www.iana.org/assignments/media-types/application/vnd.sycle+xml and a company called "Sycle" but that are IMHO edge cases and if hit these should be excluded by the user.

### CODE REVIEW

File: codespell_lib/data/dictionary.txt [hunk 1]
```diff
@@ -25863,6 +25863,7 @@ sintax->syntax
 Sionist->Zionist
 Sionists->Zionists
 siply->simply
+sircular->circular
```
Comment (peternewman) [inline codespell_lib/data/dictionary.txt:25866]: Add sircle too?

File: codespell_lib/data/dictionary.txt [hunk 2]
```diff
@@ -27647,6 +27648,11 @@ sychronizes->synchronizes
 sychronmode->synchronmode
 sychronous->synchronous
 sychronously->synchronously
+sycle->cycle
+sycled->cycled
+sycles->cycles
```
Comment (peternewman) [inline codespell_lib/data/dictionary.txt:27653]: sycling?

File: codespell_lib/data/dictionary.txt [hunk 3]
```diff
@@ -27647,6 +27648,11 @@ sychronizes->synchronizes
...
+syclic->cyclic
```
Comment (peternewman) [inline codespell_lib/data/dictionary.txt:27654]: Maybe suggest psychic too?

File: codespell_lib/data/dictionary.txt [hunk 4]
```diff
@@ -27647,6 +27648,11 @@ sychronizes->synchronizes
...
+syclical->cyclical
```
Comment (peternewman) [inline codespell_lib/data/dictionary.txt:27655]: Maybe suggest physical too?

**Review by peternewman — COMMENTED** [review]
LGTM, some possible extensions

File: codespell_lib/data/dictionary.txt (hunk 1 above)
Comment (cfi-gb) [reply codespell_lib/data/dictionary.txt:25867]: Yes, indeed. Done in 155dadf4aa0395998634ac74d94231ca1315ca85 (for all others as well).

Thanks a lot for the suggestions :+1:

File: codespell_lib/data/dictionary.txt (hunk 3 above)
Comment (cfi-gb) [reply codespell_lib/data/dictionary.txt]: 155dadf4aa0395998634ac74d94231ca1315ca85

File: codespell_lib/data/dictionary.txt (hunk 4 above)
Comment (cfi-gb) [reply codespell_lib/data/dictionary.txt]: 155dadf4aa0395998634ac74d94231ca1315ca85

File: codespell_lib/data/dictionary.txt (hunk 2 above)
Comment (cfi-gb) [reply codespell_lib/data/dictionary.txt:27654]: 155dadf4aa0395998634ac74d94231ca1315ca85

File: codespell_lib/data/dictionary.txt (hunk 1 above)
Comment (peternewman) [reply codespell_lib/data/dictionary.txt:25867]: sircles too perhaps?

**Review by peternewman — CHANGES_REQUESTED** [review]
Just one more improvement.

File: codespell_lib/data/dictionary.txt (hunk 1 above)
Comment (cfi-gb) [reply codespell_lib/data/dictionary.txt:25868]: Yes, indeed. Added in 35fbb1e52237e23ea6e110d0c3ab352e3246c8f2

**Review by peternewman — APPROVED** [review]

**Pull request merged**